openvpn/middlewares/server/auth/middleware.go

```diff
@@ -23,21 +23,26 @@ import (
 	"strings"
 )
 
-// CredentialsChecker callback checks given auth primitives (i.e. customer identity signature / node's sessionId)
-type CredentialsChecker func(username, password string) (bool, error)
-
 type middleware struct {
-	checkCredentials CredentialsChecker
-	commandWriter    management.Connection
-	currentEvent     clientEvent
+	credentialsValidator CredentialsChecker
```

Comment:
I think that instead of having more and more depenencies on each use case (validating, cleaning) maybe an event channel could be a more generic solution, and middleware will have well defined purpose - generating events from incoming client messages through management connection and passing them up

Reply:
I got the idea, but maybe its better to split this one into separate case.